src/main.rs

```diff
@@ -154,16 +154,34 @@ enum Sevctl {
 }
 
 fn download(
-    rsp: reqwest::Result<reqwest::blocking::Response>,
+    url: &str,
     usage: Usage,
 ) -> Result<sev::Certificate> {
-    let rsp = rsp.context(format!("unable to contact {} server", usage))?;
+    let mut http_request_count = 1;
+    let mut time_wait_seconds = 3;
+    let rsp = reqwest::blocking::get(url);
+    let mut rsp = rsp.context(format!("unable to contact {} server", usage))?;
+    let _status = loop {
+
+        let rsp_status = rsp.status();
+        if rsp_status.is_success() {
+            break rsp_status;
+        } else if rsp_status.is_client_error() {
+            http_request_count += 1;
+        }
+
```

Comment:
Seems like a lot of state to keep track of. Consider factoring out a helper function that takes a url as its input and returns a `reqwest::Result<reqwest::blocking::Response>`. Then you can have the initial download function call this helper and manage the retries independently from the function that actually performs the HTTP request. Does that sound reasonable?

I.e., just as an example:

```rust
use std::time::Duration;

fn make_request() -> std::io::Result<()> {
    let even = rand::random::<u32>() % 2 == 0;

    if even {
        Ok(())
    } else {
        Err(std::io::ErrorKind::TimedOut.into())
    }
}

fn main() {
    for timeout in &[0, 1, 2, 3, 4] {
        std::thread::sleep(Duration::from_secs(*timeout));

        match make_request() {
            Ok(_) => {
                println!("yay!");
                break;
            }
            Err(_) => println!("oops! retrying."),
        }
    }
}
```

I.e., in this example there's 1 function that knows exactly how to make an HTTP request, and its caller knows exactly to simply retry the request if there was an error.

Reply:
I see what you mean, will submit a change.